marconi-chain-index/src/Marconi/ChainIndex/Indexers/MintBurn.hs

```diff
@@ -292,35 +287,58 @@ type instance RI.StorableMonad MintBurnHandle = IO
 
 newtype instance RI.StorableEvent MintBurnHandle
   = MintBurnEvent TxMintEvent
+  deriving (Show)
 
 data instance RI.StorableQuery MintBurnHandle
-  = ByTxIdAndPolicyId C.TxId C.PolicyId
-  | Everything
+  = QueryByAssetId C.PolicyId C.AssetName (Maybe C.SlotNo)
+  | QueryAllMintBurn (Maybe C.SlotNo)
+  deriving (Show)
 
 newtype instance RI.StorableResult MintBurnHandle
   = MintBurnResult [TxMintRow]
+  deriving (Show)
 
 instance RI.Queryable MintBurnHandle where
-  queryStorage queryInterval memoryEvents (MintBurnHandle sqlCon _k) query = case query of
-    Everything                      -> toResult <$> sqliteSelectAll sqlCon interval
-    ByTxIdAndPolicyId txId policyId -> toResult <$> sqliteSelectByTxIdPolicyId sqlCon interval txId policyId
-    where
-      toResult storedEvents = MintBurnResult $ do
-        TxMintEvent slotNo blockHeaderHash txAssets <- storedEvents <> map coerce (toList memoryEvents)
-        (txId, mintAssets) <- NE.toList txAssets
-        MintAsset policyId assetName quantity redeemerIx redeemerData <- NE.toList mintAssets
-        pure $
-            TxMintRow
-                slotNo
-                blockHeaderHash
-                txId
-                policyId
-                assetName
-                quantity
-                redeemerIx
-                redeemerData
-
-      interval = intervalToWhereClause queryInterval
+  queryStorage queryInterval memoryEvents (MintBurnHandle sqlCon _k) query =
+      case query of
+        QueryAllMintBurn Nothing -> toResult Nothing <$> queryStoredTxMintEvents sqlCon interval
+        QueryAllMintBurn (Just slotNo) -> do
+            let conditions = interval <> (["slotNo <= :slotNo"], [":slotNo" := slotNo])
+            toResult (Just slotNo) <$> queryStoredTxMintEvents sqlCon conditions
+        QueryByAssetId policyId assetName Nothing -> do
+            let conditions = interval
+                          <> ( ["policyId = :policyId AND assetName = :assetName"]
+                             , [":policyId" := policyId, ":assetName" := assetName]
+                             )
+            toResult Nothing <$> queryStoredTxMintEvents sqlCon conditions
+        QueryByAssetId policyId assetName (Just slotNo) -> do
+            let conditions = interval
+                          <> ( ["slotNo <= :slotNo AND policyId = :policyId AND assetName = :assetName"]
```

Comment:
Should this be `slotNo = :slotNo`? Because if events up to `slotNo` are required then the interval part already does that if you add only the toSlot part. Or maybe I'm still confused as to what the interval is meant to in the indexer framework and what the jira issue is specifying.

Reply:
I'm ignoring the QInterval stuff. That should be deleted, as the semantics of it is not clear.